src/conversions.cpp

```diff
@@ -0,0 +1,127 @@
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+ */
+
+#include <moveit/drake/conversions.hpp>
+namespace moveit::drake
+{
+using ::drake::geometry::SceneGraph;
+using ::drake::multibody::AddMultibodyPlantSceneGraph;
+using ::drake::multibody::MultibodyPlant;
+using ::drake::multibody::Parser;
+using ::drake::systems::DiagramBuilder;
+
+[[nodiscard]] ::drake::trajectories::PiecewisePolynomial<double>
+getPiecewisePolynomial(const ::robot_trajectory::RobotTrajectory& robot_trajectory,
+                       const moveit::core::JointModelGroup* group)
+{
+  std::vector<double> breaks;
+  breaks.reserve(robot_trajectory.getWayPointCount());
+  std::vector<Eigen::MatrixXd> samples;
+  samples.reserve(robot_trajectory.getWayPointCount());
+
+  // Create samples & breaks
+  for (std::size_t i = 0; i < robot_trajectory.getWayPointCount(); ++i)
+  {
+    const auto& state = robot_trajectory.getWayPoint(i);
+    Eigen::VectorXd position(state.getVariableCount());
+    state.copyJointGroupPositions(group, position);
+    samples.emplace_back(position);
+    breaks.emplace_back(robot_trajectory.getWayPointDurationFromStart(i));
+  }
+
+  // Create a piecewise polynomial trajectory
+  return ::drake::trajectories::PiecewisePolynomial<double>::FirstOrderHold(breaks, samples);
+}
+
+void getRobotTrajectory(const ::drake::trajectories::Trajectory<double>& drake_trajectory, const double delta_t,
```

Comment:
Should the name contain, which way the conversion is happening? I can imagine later that we may need conversions between Drake-> moveit and moveit-> drake

Reply:
I'd like to avoid making the names too repetitive by adding the `ToMoveIt` or `ToDrake` everywhere. I tried to describe the type you're getting with the name e.g. getRobotTrajectory creates a moveit::core::RobotTrajectory